// spoonacular/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AddMealPlanTemplate200Response", {
  enumerable: true,
  get: function get() {
    return _AddMealPlanTemplate200Response["default"];
  }
});
Object.defineProperty(exports, "AddMealPlanTemplate200ResponseItemsInner", {
  enumerable: true,
  get: function get() {
    return _AddMealPlanTemplate200ResponseItemsInner["default"];
  }
});
Object.defineProperty(exports, "AddMealPlanTemplate200ResponseItemsInnerValue", {
  enumerable: true,
  get: function get() {
    return _AddMealPlanTemplate200ResponseItemsInnerValue["default"];
  }
});
Object.defineProperty(exports, "AddToMealPlanRequest", {
  enumerable: true,
  get: function get() {
    return _AddToMealPlanRequest["default"];
  }
});
Object.defineProperty(exports, "AddToMealPlanRequestValue", {
  enumerable: true,
  get: function get() {
    return _AddToMealPlanRequestValue["default"];
  }
});
Object.defineProperty(exports, "AddToMealPlanRequestValueIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _AddToMealPlanRequestValueIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "AddToShoppingListRequest", {
  enumerable: true,
  get: function get() {
    return _AddToShoppingListRequest["default"];
  }
});
Object.defineProperty(exports, "AnalyzeARecipeSearchQuery200Response", {
  enumerable: true,
  get: function get() {
    return _AnalyzeARecipeSearchQuery200Response["default"];
  }
});
Object.defineProperty(exports, "AnalyzeARecipeSearchQuery200ResponseDishesInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeARecipeSearchQuery200ResponseDishesInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeARecipeSearchQuery200ResponseIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeARecipeSearchQuery200ResponseIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeInstructions200Response", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeInstructions200Response["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeInstructions200ResponseIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeInstructions200ResponseIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeInstructions200ResponseParsedInstructionsInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeInstructions200ResponseParsedInstructionsInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInnerIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInnerIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "AnalyzeRecipeRequest", {
  enumerable: true,
  get: function get() {
    return _AnalyzeRecipeRequest["default"];
  }
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "AutocompleteIngredientSearch200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _AutocompleteIngredientSearch200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "AutocompleteProductSearch200Response", {
  enumerable: true,
  get: function get() {
    return _AutocompleteProductSearch200Response["default"];
  }
});
Object.defineProperty(exports, "AutocompleteProductSearch200ResponseResultsInner", {
  enumerable: true,
  get: function get() {
    return _AutocompleteProductSearch200ResponseResultsInner["default"];
  }
});
Object.defineProperty(exports, "AutocompleteRecipeSearch200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _AutocompleteRecipeSearch200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "ClassifyCuisine200Response", {
  enumerable: true,
  get: function get() {
    return _ClassifyCuisine200Response["default"];
  }
});
Object.defineProperty(exports, "ClassifyGroceryProduct200Response", {
  enumerable: true,
  get: function get() {
    return _ClassifyGroceryProduct200Response["default"];
  }
});
Object.defineProperty(exports, "ClassifyGroceryProductBulk200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _ClassifyGroceryProductBulk200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "ClassifyGroceryProductBulkRequestInner", {
  enumerable: true,
  get: function get() {
    return _ClassifyGroceryProductBulkRequestInner["default"];
  }
});
Object.defineProperty(exports, "ClassifyGroceryProductRequest", {
  enumerable: true,
  get: function get() {
    return _ClassifyGroceryProductRequest["default"];
  }
});
Object.defineProperty(exports, "ComparableProduct", {
  enumerable: true,
  get: function get() {
    return _ComparableProduct["default"];
  }
});
Object.defineProperty(exports, "ComputeGlycemicLoad200Response", {
  enumerable: true,
  get: function get() {
    return _ComputeGlycemicLoad200Response["default"];
  }
});
Object.defineProperty(exports, "ComputeGlycemicLoad200ResponseIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _ComputeGlycemicLoad200ResponseIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "ComputeGlycemicLoadRequest", {
  enumerable: true,
  get: function get() {
    return _ComputeGlycemicLoadRequest["default"];
  }
});
Object.defineProperty(exports, "ComputeIngredientAmount200Response", {
  enumerable: true,
  get: function get() {
    return _ComputeIngredientAmount200Response["default"];
  }
});
Object.defineProperty(exports, "ConnectUser200Response", {
  enumerable: true,
  get: function get() {
    return _ConnectUser200Response["default"];
  }
});
Object.defineProperty(exports, "ConnectUserRequest", {
  enumerable: true,
  get: function get() {
    return _ConnectUserRequest["default"];
  }
});
Object.defineProperty(exports, "ConvertAmounts200Response", {
  enumerable: true,
  get: function get() {
    return _ConvertAmounts200Response["default"];
  }
});
Object.defineProperty(exports, "CreateRecipeCard200Response", {
  enumerable: true,
  get: function get() {
    return _CreateRecipeCard200Response["default"];
  }
});
Object.defineProperty(exports, "DefaultApi", {
  enumerable: true,
  get: function get() {
    return _DefaultApi["default"];
  }
});
Object.defineProperty(exports, "DetectFoodInText200Response", {
  enumerable: true,
  get: function get() {
    return _DetectFoodInText200Response["default"];
  }
});
Object.defineProperty(exports, "DetectFoodInText200ResponseAnnotationsInner", {
  enumerable: true,
  get: function get() {
    return _DetectFoodInText200ResponseAnnotationsInner["default"];
  }
});
Object.defineProperty(exports, "GenerateMealPlan200Response", {
  enumerable: true,
  get: function get() {
    return _GenerateMealPlan200Response["default"];
  }
});
Object.defineProperty(exports, "GenerateMealPlan200ResponseNutrients", {
  enumerable: true,
  get: function get() {
    return _GenerateMealPlan200ResponseNutrients["default"];
  }
});
Object.defineProperty(exports, "GetARandomFoodJoke200Response", {
  enumerable: true,
  get: function get() {
    return _GetARandomFoodJoke200Response["default"];
  }
});
Object.defineProperty(exports, "GetAnalyzedRecipeInstructions200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _GetAnalyzedRecipeInstructions200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "GetAnalyzedRecipeInstructions200ResponseInnerStepsInner", {
  enumerable: true,
  get: function get() {
    return _GetAnalyzedRecipeInstructions200ResponseInnerStepsInner["default"];
  }
});
Object.defineProperty(exports, "GetAnalyzedRecipeInstructions200ResponseInnerStepsInnerIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _GetAnalyzedRecipeInstructions200ResponseInnerStepsInnerIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "GetComparableProducts200Response", {
  enumerable: true,
  get: function get() {
    return _GetComparableProducts200Response["default"];
  }
});
Object.defineProperty(exports, "GetComparableProducts200ResponseComparableProducts", {
  enumerable: true,
  get: function get() {
    return _GetComparableProducts200ResponseComparableProducts["default"];
  }
});
Object.defineProperty(exports, "GetConversationSuggests200Response", {
  enumerable: true,
  get: function get() {
    return _GetConversationSuggests200Response["default"];
  }
});
Object.defineProperty(exports, "GetConversationSuggests200ResponseSuggests", {
  enumerable: true,
  get: function get() {
    return _GetConversationSuggests200ResponseSuggests["default"];
  }
});
Object.defineProperty(exports, "GetConversationSuggests200ResponseSuggestsInner", {
  enumerable: true,
  get: function get() {
    return _GetConversationSuggests200ResponseSuggestsInner["default"];
  }
});
Object.defineProperty(exports, "GetDishPairingForWine200Response", {
  enumerable: true,
  get: function get() {
    return _GetDishPairingForWine200Response["default"];
  }
});
Object.defineProperty(exports, "GetIngredientSubstitutes200Response", {
  enumerable: true,
  get: function get() {
    return _GetIngredientSubstitutes200Response["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplate200Response", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplate200Response["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplate200ResponseDaysInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplate200ResponseDaysInner["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplate200ResponseDaysInnerItemsInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplate200ResponseDaysInnerItemsInner["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplate200ResponseDaysInnerItemsInnerValue", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplate200ResponseDaysInnerItemsInnerValue["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplates200Response", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplates200Response["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanTemplates200ResponseTemplatesInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanTemplates200ResponseTemplatesInner["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200Response", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200Response["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200ResponseDaysInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200ResponseDaysInner["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200ResponseDaysInnerItemsInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200ResponseDaysInnerItemsInner["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200ResponseDaysInnerItemsInnerValue", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200ResponseDaysInnerItemsInnerValue["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200ResponseDaysInnerNutritionSummary", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200ResponseDaysInnerNutritionSummary["default"];
  }
});
Object.defineProperty(exports, "GetMealPlanWeek200ResponseDaysInnerNutritionSummaryNutrientsInner", {
  enumerable: true,
  get: function get() {
    return _GetMealPlanWeek200ResponseDaysInnerNutritionSummaryNutrientsInner["default"];
  }
});
Object.defineProperty(exports, "GetRandomRecipes200Response", {
  enumerable: true,
  get: function get() {
    return _GetRandomRecipes200Response["default"];
  }
});
Object.defineProperty(exports, "GetRecipeEquipmentByID200Response", {
  enumerable: true,
  get: function get() {
    return _GetRecipeEquipmentByID200Response["default"];
  }
});
Object.defineProperty(exports, "GetRecipeEquipmentByID200ResponseEquipmentInner", {
  enumerable: true,
  get: function get() {
    return _GetRecipeEquipmentByID200ResponseEquipmentInner["default"];
  }
});
Object.defineProperty(exports, "GetRecipeIngredientsByID200Response", {
  enumerable: true,
  get: function get() {
    return _GetRecipeIngredientsByID200Response["default"];
  }
});
Object.defineProperty(exports, "GetRecipeIngredientsByID200ResponseIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _GetRecipeIngredientsByID200ResponseIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "GetRecipeNutritionWidgetByID200Response", {
  enumerable: true,
  get: function get() {
    return _GetRecipeNutritionWidgetByID200Response["default"];
  }
});
Object.defineProperty(exports, "GetRecipeNutritionWidgetByID200ResponseBadInner", {
  enumerable: true,
  get: function get() {
    return _GetRecipeNutritionWidgetByID200ResponseBadInner["default"];
  }
});
Object.defineProperty(exports, "GetRecipeNutritionWidgetByID200ResponseGoodInner", {
  enumerable: true,
  get: function get() {
    return _GetRecipeNutritionWidgetByID200ResponseGoodInner["default"];
  }
});
Object.defineProperty(exports, "GetRecipePriceBreakdownByID200Response", {
  enumerable: true,
  get: function get() {
    return _GetRecipePriceBreakdownByID200Response["default"];
  }
});
Object.defineProperty(exports, "GetRecipePriceBreakdownByID200ResponseIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _GetRecipePriceBreakdownByID200ResponseIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmount", {
  enumerable: true,
  get: function get() {
    return _GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmount["default"];
  }
});
Object.defineProperty(exports, "GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmountMetric", {
  enumerable: true,
  get: function get() {
    return _GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmountMetric["default"];
  }
});
Object.defineProperty(exports, "GetShoppingList200Response", {
  enumerable: true,
  get: function get() {
    return _GetShoppingList200Response["default"];
  }
});
Object.defineProperty(exports, "GetShoppingList200ResponseAislesInner", {
  enumerable: true,
  get: function get() {
    return _GetShoppingList200ResponseAislesInner["default"];
  }
});
Object.defineProperty(exports, "GetShoppingList200ResponseAislesInnerItemsInner", {
  enumerable: true,
  get: function get() {
    return _GetShoppingList200ResponseAislesInnerItemsInner["default"];
  }
});
Object.defineProperty(exports, "GetShoppingList200ResponseAislesInnerItemsInnerMeasures", {
  enumerable: true,
  get: function get() {
    return _GetShoppingList200ResponseAislesInnerItemsInnerMeasures["default"];
  }
});
Object.defineProperty(exports, "GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal", {
  enumerable: true,
  get: function get() {
    return _GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal["default"];
  }
});
Object.defineProperty(exports, "GetSimilarRecipes200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _GetSimilarRecipes200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "GetWineDescription200Response", {
  enumerable: true,
  get: function get() {
    return _GetWineDescription200Response["default"];
  }
});
Object.defineProperty(exports, "GetWinePairing200Response", {
  enumerable: true,
  get: function get() {
    return _GetWinePairing200Response["default"];
  }
});
Object.defineProperty(exports, "GetWinePairing200ResponseProductMatchesInner", {
  enumerable: true,
  get: function get() {
    return _GetWinePairing200ResponseProductMatchesInner["default"];
  }
});
Object.defineProperty(exports, "GetWineRecommendation200Response", {
  enumerable: true,
  get: function get() {
    return _GetWineRecommendation200Response["default"];
  }
});
Object.defineProperty(exports, "GetWineRecommendation200ResponseRecommendedWinesInner", {
  enumerable: true,
  get: function get() {
    return _GetWineRecommendation200ResponseRecommendedWinesInner["default"];
  }
});
Object.defineProperty(exports, "GuessNutritionByDishName200Response", {
  enumerable: true,
  get: function get() {
    return _GuessNutritionByDishName200Response["default"];
  }
});
Object.defineProperty(exports, "GuessNutritionByDishName200ResponseCalories", {
  enumerable: true,
  get: function get() {
    return _GuessNutritionByDishName200ResponseCalories["default"];
  }
});
Object.defineProperty(exports, "GuessNutritionByDishName200ResponseCaloriesConfidenceRange95Percent", {
  enumerable: true,
  get: function get() {
    return _GuessNutritionByDishName200ResponseCaloriesConfidenceRange95Percent["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200Response", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200Response["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200ResponseCategory", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200ResponseCategory["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200ResponseNutrition", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200ResponseNutrition["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200ResponseNutritionCalories", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200ResponseNutritionCalories["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200ResponseNutritionCaloriesConfidenceRange95Percent", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200ResponseNutritionCaloriesConfidenceRange95Percent["default"];
  }
});
Object.defineProperty(exports, "ImageAnalysisByURL200ResponseRecipesInner", {
  enumerable: true,
  get: function get() {
    return _ImageAnalysisByURL200ResponseRecipesInner["default"];
  }
});
Object.defineProperty(exports, "ImageClassificationByURL200Response", {
  enumerable: true,
  get: function get() {
    return _ImageClassificationByURL200Response["default"];
  }
});
Object.defineProperty(exports, "IngredientBasics", {
  enumerable: true,
  get: function get() {
    return _IngredientBasics["default"];
  }
});
Object.defineProperty(exports, "IngredientInformation", {
  enumerable: true,
  get: function get() {
    return _IngredientInformation["default"];
  }
});
Object.defineProperty(exports, "IngredientInformationEstimatedCost", {
  enumerable: true,
  get: function get() {
    return _IngredientInformationEstimatedCost["default"];
  }
});
Object.defineProperty(exports, "IngredientInformationNutrition", {
  enumerable: true,
  get: function get() {
    return _IngredientInformationNutrition["default"];
  }
});
Object.defineProperty(exports, "IngredientInformationNutritionPropertiesInner", {
  enumerable: true,
  get: function get() {
    return _IngredientInformationNutritionPropertiesInner["default"];
  }
});
Object.defineProperty(exports, "IngredientSearch200Response", {
  enumerable: true,
  get: function get() {
    return _IngredientSearch200Response["default"];
  }
});
Object.defineProperty(exports, "IngredientSearch200ResponseResultsInner", {
  enumerable: true,
  get: function get() {
    return _IngredientSearch200ResponseResultsInner["default"];
  }
});
Object.defineProperty(exports, "IngredientsApi", {
  enumerable: true,
  get: function get() {
    return _IngredientsApi["default"];
  }
});
Object.defineProperty(exports, "MapIngredientsToGroceryProducts200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _MapIngredientsToGroceryProducts200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "MapIngredientsToGroceryProducts200ResponseInnerProductsInner", {
  enumerable: true,
  get: function get() {
    return _MapIngredientsToGroceryProducts200ResponseInnerProductsInner["default"];
  }
});
Object.defineProperty(exports, "MapIngredientsToGroceryProductsRequest", {
  enumerable: true,
  get: function get() {
    return _MapIngredientsToGroceryProductsRequest["default"];
  }
});
Object.defineProperty(exports, "MealPlanningApi", {
  enumerable: true,
  get: function get() {
    return _MealPlanningApi["default"];
  }
});
Object.defineProperty(exports, "MenuItem", {
  enumerable: true,
  get: function get() {
    return _MenuItem["default"];
  }
});
Object.defineProperty(exports, "MenuItemServings", {
  enumerable: true,
  get: function get() {
    return _MenuItemServings["default"];
  }
});
Object.defineProperty(exports, "MenuItemsApi", {
  enumerable: true,
  get: function get() {
    return _MenuItemsApi["default"];
  }
});
Object.defineProperty(exports, "MiscApi", {
  enumerable: true,
  get: function get() {
    return _MiscApi["default"];
  }
});
Object.defineProperty(exports, "ProductInformation", {
  enumerable: true,
  get: function get() {
    return _ProductInformation["default"];
  }
});
Object.defineProperty(exports, "ProductInformationCredits", {
  enumerable: true,
  get: function get() {
    return _ProductInformationCredits["default"];
  }
});
Object.defineProperty(exports, "ProductsApi", {
  enumerable: true,
  get: function get() {
    return _ProductsApi["default"];
  }
});
Object.defineProperty(exports, "QuickAnswer200Response", {
  enumerable: true,
  get: function get() {
    return _QuickAnswer200Response["default"];
  }
});
Object.defineProperty(exports, "RecipeInformation", {
  enumerable: true,
  get: function get() {
    return _RecipeInformation["default"];
  }
});
Object.defineProperty(exports, "RecipeInformationExtendedIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _RecipeInformationExtendedIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "RecipeInformationExtendedIngredientsInnerMeasures", {
  enumerable: true,
  get: function get() {
    return _RecipeInformationExtendedIngredientsInnerMeasures["default"];
  }
});
Object.defineProperty(exports, "RecipeInformationExtendedIngredientsInnerMeasuresMetric", {
  enumerable: true,
  get: function get() {
    return _RecipeInformationExtendedIngredientsInnerMeasuresMetric["default"];
  }
});
Object.defineProperty(exports, "RecipeInformationWinePairing", {
  enumerable: true,
  get: function get() {
    return _RecipeInformationWinePairing["default"];
  }
});
Object.defineProperty(exports, "RecipeInformationWinePairingProductMatchesInner", {
  enumerable: true,
  get: function get() {
    return _RecipeInformationWinePairingProductMatchesInner["default"];
  }
});
Object.defineProperty(exports, "RecipesApi", {
  enumerable: true,
  get: function get() {
    return _RecipesApi["default"];
  }
});
Object.defineProperty(exports, "SearchAllFood200Response", {
  enumerable: true,
  get: function get() {
    return _SearchAllFood200Response["default"];
  }
});
Object.defineProperty(exports, "SearchAllFood200ResponseSearchResultsInner", {
  enumerable: true,
  get: function get() {
    return _SearchAllFood200ResponseSearchResultsInner["default"];
  }
});
Object.defineProperty(exports, "SearchCustomFoods200Response", {
  enumerable: true,
  get: function get() {
    return _SearchCustomFoods200Response["default"];
  }
});
Object.defineProperty(exports, "SearchCustomFoods200ResponseCustomFoodsInner", {
  enumerable: true,
  get: function get() {
    return _SearchCustomFoods200ResponseCustomFoodsInner["default"];
  }
});
Object.defineProperty(exports, "SearchFoodVideos200Response", {
  enumerable: true,
  get: function get() {
    return _SearchFoodVideos200Response["default"];
  }
});
Object.defineProperty(exports, "SearchFoodVideos200ResponseVideosInner", {
  enumerable: true,
  get: function get() {
    return _SearchFoodVideos200ResponseVideosInner["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProducts200Response", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProducts200Response["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProductsByUPC200Response", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProductsByUPC200Response["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProductsByUPC200ResponseNutrition", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProductsByUPC200ResponseNutrition["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner["default"];
  }
});
Object.defineProperty(exports, "SearchGroceryProductsByUPC200ResponseServings", {
  enumerable: true,
  get: function get() {
    return _SearchGroceryProductsByUPC200ResponseServings["default"];
  }
});
Object.defineProperty(exports, "SearchMenuItems200Response", {
  enumerable: true,
  get: function get() {
    return _SearchMenuItems200Response["default"];
  }
});
Object.defineProperty(exports, "SearchRecipes200Response", {
  enumerable: true,
  get: function get() {
    return _SearchRecipes200Response["default"];
  }
});
Object.defineProperty(exports, "SearchRecipes200ResponseResultsInner", {
  enumerable: true,
  get: function get() {
    return _SearchRecipes200ResponseResultsInner["default"];
  }
});
Object.defineProperty(exports, "SearchRecipesByIngredients200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _SearchRecipesByIngredients200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner", {
  enumerable: true,
  get: function get() {
    return _SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner["default"];
  }
});
Object.defineProperty(exports, "SearchRecipesByNutrients200ResponseInner", {
  enumerable: true,
  get: function get() {
    return _SearchRecipesByNutrients200ResponseInner["default"];
  }
});
Object.defineProperty(exports, "SearchRestaurants200Response", {
  enumerable: true,
  get: function get() {
    return _SearchRestaurants200Response["default"];
  }
});
Object.defineProperty(exports, "SearchRestaurants200ResponseRestaurantsInner", {
  enumerable: true,
  get: function get() {
    return _SearchRestaurants200ResponseRestaurantsInner["default"];
  }
});
Object.defineProperty(exports, "SearchRestaurants200ResponseRestaurantsInnerAddress", {
  enumerable: true,
  get: function get() {
    return _SearchRestaurants200ResponseRestaurantsInnerAddress["default"];
  }
});
Object.defineProperty(exports, "SearchRestaurants200ResponseRestaurantsInnerLocalHours", {
  enumerable: true,
  get: function get() {
    return _SearchRestaurants200ResponseRestaurantsInnerLocalHours["default"];
  }
});
Object.defineProperty(exports, "SearchRestaurants200ResponseRestaurantsInnerLocalHoursOperational", {
  enumerable: true,
  get: function get() {
    return _SearchRestaurants200ResponseRestaurantsInnerLocalHoursOperational["default"];
  }
});
Object.defineProperty(exports, "SearchResult", {
  enumerable: true,
  get: function get() {
    return _SearchResult["default"];
  }
});
Object.defineProperty(exports, "SearchSiteContent200Response", {
  enumerable: true,
  get: function get() {
    return _SearchSiteContent200Response["default"];
  }
});
Object.defineProperty(exports, "SummarizeRecipe200Response", {
  enumerable: true,
  get: function get() {
    return _SummarizeRecipe200Response["default"];
  }
});
Object.defineProperty(exports, "TalkToChatbot200Response", {
  enumerable: true,
  get: function get() {
    return _TalkToChatbot200Response["default"];
  }
});
Object.defineProperty(exports, "TalkToChatbot200ResponseMediaInner", {
  enumerable: true,
  get: function get() {
    return _TalkToChatbot200ResponseMediaInner["default"];
  }
});
Object.defineProperty(exports, "TasteInformation", {
  enumerable: true,
  get: function get() {
    return _TasteInformation["default"];
  }
});
Object.defineProperty(exports, "WineApi", {
  enumerable: true,
  get: function get() {
    return _WineApi["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _AddMealPlanTemplate200Response = _interopRequireDefault(require("./model/AddMealPlanTemplate200Response"));
var _AddMealPlanTemplate200ResponseItemsInner = _interopRequireDefault(require("./model/AddMealPlanTemplate200ResponseItemsInner"));
var _AddMealPlanTemplate200ResponseItemsInnerValue = _interopRequireDefault(require("./model/AddMealPlanTemplate200ResponseItemsInnerValue"));
var _AddToMealPlanRequest = _interopRequireDefault(require("./model/AddToMealPlanRequest"));
var _AddToMealPlanRequestValue = _interopRequireDefault(require("./model/AddToMealPlanRequestValue"));
var _AddToMealPlanRequestValueIngredientsInner = _interopRequireDefault(require("./model/AddToMealPlanRequestValueIngredientsInner"));
var _AddToShoppingListRequest = _interopRequireDefault(require("./model/AddToShoppingListRequest"));
var _AnalyzeARecipeSearchQuery200Response = _interopRequireDefault(require("./model/AnalyzeARecipeSearchQuery200Response"));
var _AnalyzeARecipeSearchQuery200ResponseDishesInner = _interopRequireDefault(require("./model/AnalyzeARecipeSearchQuery200ResponseDishesInner"));
var _AnalyzeARecipeSearchQuery200ResponseIngredientsInner = _interopRequireDefault(require("./model/AnalyzeARecipeSearchQuery200ResponseIngredientsInner"));
var _AnalyzeRecipeInstructions200Response = _interopRequireDefault(require("./model/AnalyzeRecipeInstructions200Response"));
var _AnalyzeRecipeInstructions200ResponseIngredientsInner = _interopRequireDefault(require("./model/AnalyzeRecipeInstructions200ResponseIngredientsInner"));
var _AnalyzeRecipeInstructions200ResponseParsedInstructionsInner = _interopRequireDefault(require("./model/AnalyzeRecipeInstructions200ResponseParsedInstructionsInner"));
var _AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInner = _interopRequireDefault(require("./model/AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInner"));
var _AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInnerIngredientsInner = _interopRequireDefault(require("./model/AnalyzeRecipeInstructions200ResponseParsedInstructionsInnerStepsInnerIngredientsInner"));
var _AnalyzeRecipeRequest = _interopRequireDefault(require("./model/AnalyzeRecipeRequest"));
var _AutocompleteIngredientSearch200ResponseInner = _interopRequireDefault(require("./model/AutocompleteIngredientSearch200ResponseInner"));
var _AutocompleteProductSearch200Response = _interopRequireDefault(require("./model/AutocompleteProductSearch200Response"));
var _AutocompleteProductSearch200ResponseResultsInner = _interopRequireDefault(require("./model/AutocompleteProductSearch200ResponseResultsInner"));
var _AutocompleteRecipeSearch200ResponseInner = _interopRequireDefault(require("./model/AutocompleteRecipeSearch200ResponseInner"));
var _ClassifyCuisine200Response = _interopRequireDefault(require("./model/ClassifyCuisine200Response"));
var _ClassifyGroceryProduct200Response = _interopRequireDefault(require("./model/ClassifyGroceryProduct200Response"));
var _ClassifyGroceryProductBulk200ResponseInner = _interopRequireDefault(require("./model/ClassifyGroceryProductBulk200ResponseInner"));
var _ClassifyGroceryProductBulkRequestInner = _interopRequireDefault(require("./model/ClassifyGroceryProductBulkRequestInner"));
var _ClassifyGroceryProductRequest = _interopRequireDefault(require("./model/ClassifyGroceryProductRequest"));
var _ComparableProduct = _interopRequireDefault(require("./model/ComparableProduct"));
var _ComputeGlycemicLoad200Response = _interopRequireDefault(require("./model/ComputeGlycemicLoad200Response"));
var _ComputeGlycemicLoad200ResponseIngredientsInner = _interopRequireDefault(require("./model/ComputeGlycemicLoad200ResponseIngredientsInner"));
var _ComputeGlycemicLoadRequest = _interopRequireDefault(require("./model/ComputeGlycemicLoadRequest"));
var _ComputeIngredientAmount200Response = _interopRequireDefault(require("./model/ComputeIngredientAmount200Response"));
var _ConnectUser200Response = _interopRequireDefault(require("./model/ConnectUser200Response"));
var _ConnectUserRequest = _interopRequireDefault(require("./model/ConnectUserRequest"));
var _ConvertAmounts200Response = _interopRequireDefault(require("./model/ConvertAmounts200Response"));
var _CreateRecipeCard200Response = _interopRequireDefault(require("./model/CreateRecipeCard200Response"));
var _DetectFoodInText200Response = _interopRequireDefault(require("./model/DetectFoodInText200Response"));
var _DetectFoodInText200ResponseAnnotationsInner = _interopRequireDefault(require("./model/DetectFoodInText200ResponseAnnotationsInner"));
var _GenerateMealPlan200Response = _interopRequireDefault(require("./model/GenerateMealPlan200Response"));
var _GenerateMealPlan200ResponseNutrients = _interopRequireDefault(require("./model/GenerateMealPlan200ResponseNutrients"));
var _GetARandomFoodJoke200Response = _interopRequireDefault(require("./model/GetARandomFoodJoke200Response"));
var _GetAnalyzedRecipeInstructions200ResponseInner = _interopRequireDefault(require("./model/GetAnalyzedRecipeInstructions200ResponseInner"));
var _GetAnalyzedRecipeInstructions200ResponseInnerStepsInner = _interopRequireDefault(require("./model/GetAnalyzedRecipeInstructions200ResponseInnerStepsInner"));
var _GetAnalyzedRecipeInstructions200ResponseInnerStepsInnerIngredientsInner = _interopRequireDefault(require("./model/GetAnalyzedRecipeInstructions200ResponseInnerStepsInnerIngredientsInner"));
var _GetComparableProducts200Response = _interopRequireDefault(require("./model/GetComparableProducts200Response"));
var _GetComparableProducts200ResponseComparableProducts = _interopRequireDefault(require("./model/GetComparableProducts200ResponseComparableProducts"));
var _GetConversationSuggests200Response = _interopRequireDefault(require("./model/GetConversationSuggests200Response"));
var _GetConversationSuggests200ResponseSuggests = _interopRequireDefault(require("./model/GetConversationSuggests200ResponseSuggests"));
var _GetConversationSuggests200ResponseSuggestsInner = _interopRequireDefault(require("./model/GetConversationSuggests200ResponseSuggestsInner"));
var _GetDishPairingForWine200Response = _interopRequireDefault(require("./model/GetDishPairingForWine200Response"));
var _GetIngredientSubstitutes200Response = _interopRequireDefault(require("./model/GetIngredientSubstitutes200Response"));
var _GetMealPlanTemplate200Response = _interopRequireDefault(require("./model/GetMealPlanTemplate200Response"));
var _GetMealPlanTemplate200ResponseDaysInner = _interopRequireDefault(require("./model/GetMealPlanTemplate200ResponseDaysInner"));
var _GetMealPlanTemplate200ResponseDaysInnerItemsInner = _interopRequireDefault(require("./model/GetMealPlanTemplate200ResponseDaysInnerItemsInner"));
var _GetMealPlanTemplate200ResponseDaysInnerItemsInnerValue = _interopRequireDefault(require("./model/GetMealPlanTemplate200ResponseDaysInnerItemsInnerValue"));
var _GetMealPlanTemplates200Response = _interopRequireDefault(require("./model/GetMealPlanTemplates200Response"));
var _GetMealPlanTemplates200ResponseTemplatesInner = _interopRequireDefault(require("./model/GetMealPlanTemplates200ResponseTemplatesInner"));
var _GetMealPlanWeek200Response = _interopRequireDefault(require("./model/GetMealPlanWeek200Response"));
var _GetMealPlanWeek200ResponseDaysInner = _interopRequireDefault(require("./model/GetMealPlanWeek200ResponseDaysInner"));
var _GetMealPlanWeek200ResponseDaysInnerItemsInner = _interopRequireDefault(require("./model/GetMealPlanWeek200ResponseDaysInnerItemsInner"));
var _GetMealPlanWeek200ResponseDaysInnerItemsInnerValue = _interopRequireDefault(require("./model/GetMealPlanWeek200ResponseDaysInnerItemsInnerValue"));
var _GetMealPlanWeek200ResponseDaysInnerNutritionSummary = _interopRequireDefault(require("./model/GetMealPlanWeek200ResponseDaysInnerNutritionSummary"));
var _GetMealPlanWeek200ResponseDaysInnerNutritionSummaryNutrientsInner = _interopRequireDefault(require("./model/GetMealPlanWeek200ResponseDaysInnerNutritionSummaryNutrientsInner"));
var _GetRandomRecipes200Response = _interopRequireDefault(require("./model/GetRandomRecipes200Response"));
var _GetRecipeEquipmentByID200Response = _interopRequireDefault(require("./model/GetRecipeEquipmentByID200Response"));
var _GetRecipeEquipmentByID200ResponseEquipmentInner = _interopRequireDefault(require("./model/GetRecipeEquipmentByID200ResponseEquipmentInner"));
var _GetRecipeIngredientsByID200Response = _interopRequireDefault(require("./model/GetRecipeIngredientsByID200Response"));
var _GetRecipeIngredientsByID200ResponseIngredientsInner = _interopRequireDefault(require("./model/GetRecipeIngredientsByID200ResponseIngredientsInner"));
var _GetRecipeNutritionWidgetByID200Response = _interopRequireDefault(require("./model/GetRecipeNutritionWidgetByID200Response"));
var _GetRecipeNutritionWidgetByID200ResponseBadInner = _interopRequireDefault(require("./model/GetRecipeNutritionWidgetByID200ResponseBadInner"));
var _GetRecipeNutritionWidgetByID200ResponseGoodInner = _interopRequireDefault(require("./model/GetRecipeNutritionWidgetByID200ResponseGoodInner"));
var _GetRecipePriceBreakdownByID200Response = _interopRequireDefault(require("./model/GetRecipePriceBreakdownByID200Response"));
var _GetRecipePriceBreakdownByID200ResponseIngredientsInner = _interopRequireDefault(require("./model/GetRecipePriceBreakdownByID200ResponseIngredientsInner"));
var _GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmount = _interopRequireDefault(require("./model/GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmount"));
var _GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmountMetric = _interopRequireDefault(require("./model/GetRecipePriceBreakdownByID200ResponseIngredientsInnerAmountMetric"));
var _GetShoppingList200Response = _interopRequireDefault(require("./model/GetShoppingList200Response"));
var _GetShoppingList200ResponseAislesInner = _interopRequireDefault(require("./model/GetShoppingList200ResponseAislesInner"));
var _GetShoppingList200ResponseAislesInnerItemsInner = _interopRequireDefault(require("./model/GetShoppingList200ResponseAislesInnerItemsInner"));
var _GetShoppingList200ResponseAislesInnerItemsInnerMeasures = _interopRequireDefault(require("./model/GetShoppingList200ResponseAislesInnerItemsInnerMeasures"));
var _GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal = _interopRequireDefault(require("./model/GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal"));
var _GetSimilarRecipes200ResponseInner = _interopRequireDefault(require("./model/GetSimilarRecipes200ResponseInner"));
var _GetWineDescription200Response = _interopRequireDefault(require("./model/GetWineDescription200Response"));
var _GetWinePairing200Response = _interopRequireDefault(require("./model/GetWinePairing200Response"));
var _GetWinePairing200ResponseProductMatchesInner = _interopRequireDefault(require("./model/GetWinePairing200ResponseProductMatchesInner"));
var _GetWineRecommendation200Response = _interopRequireDefault(require("./model/GetWineRecommendation200Response"));
var _GetWineRecommendation200ResponseRecommendedWinesInner = _interopRequireDefault(require("./model/GetWineRecommendation200ResponseRecommendedWinesInner"));
var _GuessNutritionByDishName200Response = _interopRequireDefault(require("./model/GuessNutritionByDishName200Response"));
var _GuessNutritionByDishName200ResponseCalories = _interopRequireDefault(require("./model/GuessNutritionByDishName200ResponseCalories"));
var _GuessNutritionByDishName200ResponseCaloriesConfidenceRange95Percent = _interopRequireDefault(require("./model/GuessNutritionByDishName200ResponseCaloriesConfidenceRange95Percent"));
var _ImageAnalysisByURL200Response = _interopRequireDefault(require("./model/ImageAnalysisByURL200Response"));
var _ImageAnalysisByURL200ResponseCategory = _interopRequireDefault(require("./model/ImageAnalysisByURL200ResponseCategory"));
var _ImageAnalysisByURL200ResponseNutrition = _interopRequireDefault(require("./model/ImageAnalysisByURL200ResponseNutrition"));
var _ImageAnalysisByURL200ResponseNutritionCalories = _interopRequireDefault(require("./model/ImageAnalysisByURL200ResponseNutritionCalories"));
var _ImageAnalysisByURL200ResponseNutritionCaloriesConfidenceRange95Percent = _interopRequireDefault(require("./model/ImageAnalysisByURL200ResponseNutritionCaloriesConfidenceRange95Percent"));
var _ImageAnalysisByURL200ResponseRecipesInner = _interopRequireDefault(require("./model/ImageAnalysisByURL200ResponseRecipesInner"));
var _ImageClassificationByURL200Response = _interopRequireDefault(require("./model/ImageClassificationByURL200Response"));
var _IngredientBasics = _interopRequireDefault(require("./model/IngredientBasics"));
var _IngredientInformation = _interopRequireDefault(require("./model/IngredientInformation"));
var _IngredientInformationEstimatedCost = _interopRequireDefault(require("./model/IngredientInformationEstimatedCost"));
var _IngredientInformationNutrition = _interopRequireDefault(require("./model/IngredientInformationNutrition"));
var _IngredientInformationNutritionPropertiesInner = _interopRequireDefault(require("./model/IngredientInformationNutritionPropertiesInner"));
var _IngredientSearch200Response = _interopRequireDefault(require("./model/IngredientSearch200Response"));
var _IngredientSearch200ResponseResultsInner = _interopRequireDefault(require("./model/IngredientSearch200ResponseResultsInner"));
var _MapIngredientsToGroceryProducts200ResponseInner = _interopRequireDefault(require("./model/MapIngredientsToGroceryProducts200ResponseInner"));
var _MapIngredientsToGroceryProducts200ResponseInnerProductsInner = _interopRequireDefault(require("./model/MapIngredientsToGroceryProducts200ResponseInnerProductsInner"));
var _MapIngredientsToGroceryProductsRequest = _interopRequireDefault(require("./model/MapIngredientsToGroceryProductsRequest"));
var _MenuItem = _interopRequireDefault(require("./model/MenuItem"));
var _MenuItemServings = _interopRequireDefault(require("./model/MenuItemServings"));
var _ProductInformation = _interopRequireDefault(require("./model/ProductInformation"));
var _ProductInformationCredits = _interopRequireDefault(require("./model/ProductInformationCredits"));
var _QuickAnswer200Response = _interopRequireDefault(require("./model/QuickAnswer200Response"));
var _RecipeInformation = _interopRequireDefault(require("./model/RecipeInformation"));
var _RecipeInformationExtendedIngredientsInner = _interopRequireDefault(require("./model/RecipeInformationExtendedIngredientsInner"));
var _RecipeInformationExtendedIngredientsInnerMeasures = _interopRequireDefault(require("./model/RecipeInformationExtendedIngredientsInnerMeasures"));
var _RecipeInformationExtendedIngredientsInnerMeasuresMetric = _interopRequireDefault(require("./model/RecipeInformationExtendedIngredientsInnerMeasuresMetric"));
var _RecipeInformationWinePairing = _interopRequireDefault(require("./model/RecipeInformationWinePairing"));
var _RecipeInformationWinePairingProductMatchesInner = _interopRequireDefault(require("./model/RecipeInformationWinePairingProductMatchesInner"));
var _SearchAllFood200Response = _interopRequireDefault(require("./model/SearchAllFood200Response"));
var _SearchAllFood200ResponseSearchResultsInner = _interopRequireDefault(require("./model/SearchAllFood200ResponseSearchResultsInner"));
var _SearchCustomFoods200Response = _interopRequireDefault(require("./model/SearchCustomFoods200Response"));
var _SearchCustomFoods200ResponseCustomFoodsInner = _interopRequireDefault(require("./model/SearchCustomFoods200ResponseCustomFoodsInner"));
var _SearchFoodVideos200Response = _interopRequireDefault(require("./model/SearchFoodVideos200Response"));
var _SearchFoodVideos200ResponseVideosInner = _interopRequireDefault(require("./model/SearchFoodVideos200ResponseVideosInner"));
var _SearchGroceryProducts200Response = _interopRequireDefault(require("./model/SearchGroceryProducts200Response"));
var _SearchGroceryProductsByUPC200Response = _interopRequireDefault(require("./model/SearchGroceryProductsByUPC200Response"));
var _SearchGroceryProductsByUPC200ResponseNutrition = _interopRequireDefault(require("./model/SearchGroceryProductsByUPC200ResponseNutrition"));
var _SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown = _interopRequireDefault(require("./model/SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown"));
var _SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner = _interopRequireDefault(require("./model/SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner"));
var _SearchGroceryProductsByUPC200ResponseServings = _interopRequireDefault(require("./model/SearchGroceryProductsByUPC200ResponseServings"));
var _SearchMenuItems200Response = _interopRequireDefault(require("./model/SearchMenuItems200Response"));
var _SearchRecipes200Response = _interopRequireDefault(require("./model/SearchRecipes200Response"));
var _SearchRecipes200ResponseResultsInner = _interopRequireDefault(require("./model/SearchRecipes200ResponseResultsInner"));
var _SearchRecipesByIngredients200ResponseInner = _interopRequireDefault(require("./model/SearchRecipesByIngredients200ResponseInner"));
var _SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner = _interopRequireDefault(require("./model/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner"));
var _SearchRecipesByNutrients200ResponseInner = _interopRequireDefault(require("./model/SearchRecipesByNutrients200ResponseInner"));
var _SearchRestaurants200Response = _interopRequireDefault(require("./model/SearchRestaurants200Response"));
var _SearchRestaurants200ResponseRestaurantsInner = _interopRequireDefault(require("./model/SearchRestaurants200ResponseRestaurantsInner"));
var _SearchRestaurants200ResponseRestaurantsInnerAddress = _interopRequireDefault(require("./model/SearchRestaurants200ResponseRestaurantsInnerAddress"));
var _SearchRestaurants200ResponseRestaurantsInnerLocalHours = _interopRequireDefault(require("./model/SearchRestaurants200ResponseRestaurantsInnerLocalHours"));
var _SearchRestaurants200ResponseRestaurantsInnerLocalHoursOperational = _interopRequireDefault(require("./model/SearchRestaurants200ResponseRestaurantsInnerLocalHoursOperational"));
var _SearchResult = _interopRequireDefault(require("./model/SearchResult"));
var _SearchSiteContent200Response = _interopRequireDefault(require("./model/SearchSiteContent200Response"));
var _SummarizeRecipe200Response = _interopRequireDefault(require("./model/SummarizeRecipe200Response"));
var _TalkToChatbot200Response = _interopRequireDefault(require("./model/TalkToChatbot200Response"));
var _TalkToChatbot200ResponseMediaInner = _interopRequireDefault(require("./model/TalkToChatbot200ResponseMediaInner"));
var _TasteInformation = _interopRequireDefault(require("./model/TasteInformation"));
var _DefaultApi = _interopRequireDefault(require("./api/DefaultApi"));
var _IngredientsApi = _interopRequireDefault(require("./api/IngredientsApi"));
var _MealPlanningApi = _interopRequireDefault(require("./api/MealPlanningApi"));
var _MenuItemsApi = _interopRequireDefault(require("./api/MenuItemsApi"));
var _MiscApi = _interopRequireDefault(require("./api/MiscApi"));
var _ProductsApi = _interopRequireDefault(require("./api/ProductsApi"));
var _RecipesApi = _interopRequireDefault(require("./api/RecipesApi"));
var _WineApi = _interopRequireDefault(require("./api/WineApi"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
