// spoonacular/dist/model/IngredientInformationNutrition.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal = _interopRequireDefault(require("./GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal"));
var _IngredientInformationNutritionPropertiesInner = _interopRequireDefault(require("./IngredientInformationNutritionPropertiesInner"));
var _SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown = _interopRequireDefault(require("./SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown"));
var _SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner = _interopRequireDefault(require("./SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The IngredientInformationNutrition model module.
 * @module model/IngredientInformationNutrition
 * @version 2.0.2
 */
var IngredientInformationNutrition = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>IngredientInformationNutrition</code>.
   * @alias module:model/IngredientInformationNutrition
   * @param nutrients {Array.<module:model/SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner>} 
   * @param properties {Array.<module:model/IngredientInformationNutritionPropertiesInner>} 
   * @param caloricBreakdown {module:model/SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown} 
   * @param weightPerServing {module:model/GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal} 
   */
  function IngredientInformationNutrition(nutrients, properties, caloricBreakdown, weightPerServing) {
    _classCallCheck(this, IngredientInformationNutrition);
    IngredientInformationNutrition.initialize(this, nutrients, properties, caloricBreakdown, weightPerServing);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(IngredientInformationNutrition, null, [{
    key: "initialize",
    value: function initialize(obj, nutrients, properties, caloricBreakdown, weightPerServing) {
      obj['nutrients'] = nutrients;
      obj['properties'] = properties;
      obj['caloricBreakdown'] = caloricBreakdown;
      obj['weightPerServing'] = weightPerServing;
    }

    /**
     * Constructs a <code>IngredientInformationNutrition</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/IngredientInformationNutrition} obj Optional instance to populate.
     * @return {module:model/IngredientInformationNutrition} The populated <code>IngredientInformationNutrition</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new IngredientInformationNutrition();
        if (data.hasOwnProperty('nutrients')) {
          obj['nutrients'] = _ApiClient["default"].convertToType(data['nutrients'], [_SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner["default"]]);
        }
        if (data.hasOwnProperty('properties')) {
          obj['properties'] = _ApiClient["default"].convertToType(data['properties'], [_IngredientInformationNutritionPropertiesInner["default"]]);
        }
        if (data.hasOwnProperty('caloricBreakdown')) {
          obj['caloricBreakdown'] = _SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown["default"].constructFromObject(data['caloricBreakdown']);
        }
        if (data.hasOwnProperty('weightPerServing')) {
          obj['weightPerServing'] = _GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal["default"].constructFromObject(data['weightPerServing']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>IngredientInformationNutrition</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>IngredientInformationNutrition</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(IngredientInformationNutrition.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['nutrients']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['nutrients'])) {
          throw new Error("Expected the field `nutrients` to be an array in the JSON data but got " + data['nutrients']);
        }
        // validate the optional field `nutrients` (array)
        var _iterator2 = _createForOfIteratorHelper(data['nutrients']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      if (data['properties']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['properties'])) {
          throw new Error("Expected the field `properties` to be an array in the JSON data but got " + data['properties']);
        }
        // validate the optional field `properties` (array)
        var _iterator3 = _createForOfIteratorHelper(data['properties']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _IngredientInformationNutritionPropertiesInner["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      // validate the optional field `caloricBreakdown`
      if (data['caloricBreakdown']) {
        // data not null
        _SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown["default"].validateJSON(data['caloricBreakdown']);
      }
      // validate the optional field `weightPerServing`
      if (data['weightPerServing']) {
        // data not null
        _GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal["default"].validateJSON(data['weightPerServing']);
      }
      return true;
    }
  }]);
}();
IngredientInformationNutrition.RequiredProperties = ["nutrients", "properties", "caloricBreakdown", "weightPerServing"];

/**
 * @member {Array.<module:model/SearchGroceryProductsByUPC200ResponseNutritionNutrientsInner>} nutrients
 */
IngredientInformationNutrition.prototype['nutrients'] = undefined;

/**
 * @member {Array.<module:model/IngredientInformationNutritionPropertiesInner>} properties
 */
IngredientInformationNutrition.prototype['properties'] = undefined;

/**
 * @member {module:model/SearchGroceryProductsByUPC200ResponseNutritionCaloricBreakdown} caloricBreakdown
 */
IngredientInformationNutrition.prototype['caloricBreakdown'] = undefined;

/**
 * @member {module:model/GetShoppingList200ResponseAislesInnerItemsInnerMeasuresOriginal} weightPerServing
 */
IngredientInformationNutrition.prototype['weightPerServing'] = undefined;
var _default = exports["default"] = IngredientInformationNutrition;
