// spoonacular/dist/model/SearchSiteContent200Response.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SearchResult = _interopRequireDefault(require("./SearchResult"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The SearchSiteContent200Response model module.
 * @module model/SearchSiteContent200Response
 * @version 2.0.2
 */
var SearchSiteContent200Response = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SearchSiteContent200Response</code>.
   * 
   * @alias module:model/SearchSiteContent200Response
   * @param articles {Array.<module:model/SearchResult>} 
   * @param groceryProducts {Array.<module:model/SearchResult>} 
   * @param menuItems {Array.<module:model/SearchResult>} 
   * @param recipes {Array.<module:model/SearchResult>} 
   */
  function SearchSiteContent200Response(articles, groceryProducts, menuItems, recipes) {
    _classCallCheck(this, SearchSiteContent200Response);
    SearchSiteContent200Response.initialize(this, articles, groceryProducts, menuItems, recipes);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(SearchSiteContent200Response, null, [{
    key: "initialize",
    value: function initialize(obj, articles, groceryProducts, menuItems, recipes) {
      obj['Articles'] = articles;
      obj['Grocery Products'] = groceryProducts;
      obj['Menu Items'] = menuItems;
      obj['Recipes'] = recipes;
    }

    /**
     * Constructs a <code>SearchSiteContent200Response</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SearchSiteContent200Response} obj Optional instance to populate.
     * @return {module:model/SearchSiteContent200Response} The populated <code>SearchSiteContent200Response</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SearchSiteContent200Response();
        if (data.hasOwnProperty('Articles')) {
          obj['Articles'] = _ApiClient["default"].convertToType(data['Articles'], [_SearchResult["default"]]);
        }
        if (data.hasOwnProperty('Grocery Products')) {
          obj['Grocery Products'] = _ApiClient["default"].convertToType(data['Grocery Products'], [_SearchResult["default"]]);
        }
        if (data.hasOwnProperty('Menu Items')) {
          obj['Menu Items'] = _ApiClient["default"].convertToType(data['Menu Items'], [_SearchResult["default"]]);
        }
        if (data.hasOwnProperty('Recipes')) {
          obj['Recipes'] = _ApiClient["default"].convertToType(data['Recipes'], [_SearchResult["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SearchSiteContent200Response</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SearchSiteContent200Response</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(SearchSiteContent200Response.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['Articles']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['Articles'])) {
          throw new Error("Expected the field `Articles` to be an array in the JSON data but got " + data['Articles']);
        }
        // validate the optional field `Articles` (array)
        var _iterator2 = _createForOfIteratorHelper(data['Articles']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _SearchResult["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      if (data['Grocery Products']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['Grocery Products'])) {
          throw new Error("Expected the field `Grocery Products` to be an array in the JSON data but got " + data['Grocery Products']);
        }
        // validate the optional field `Grocery Products` (array)
        var _iterator3 = _createForOfIteratorHelper(data['Grocery Products']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _SearchResult["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      if (data['Menu Items']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['Menu Items'])) {
          throw new Error("Expected the field `Menu Items` to be an array in the JSON data but got " + data['Menu Items']);
        }
        // validate the optional field `Menu Items` (array)
        var _iterator4 = _createForOfIteratorHelper(data['Menu Items']),
          _step4;
        try {
          for (_iterator4.s(); !(_step4 = _iterator4.n()).done;) {
            var _item2 = _step4.value;
            _SearchResult["default"].validateJSON(_item2);
          }
        } catch (err) {
          _iterator4.e(err);
        } finally {
          _iterator4.f();
        }
        ;
      }
      if (data['Recipes']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['Recipes'])) {
          throw new Error("Expected the field `Recipes` to be an array in the JSON data but got " + data['Recipes']);
        }
        // validate the optional field `Recipes` (array)
        var _iterator5 = _createForOfIteratorHelper(data['Recipes']),
          _step5;
        try {
          for (_iterator5.s(); !(_step5 = _iterator5.n()).done;) {
            var _item3 = _step5.value;
            _SearchResult["default"].validateJSON(_item3);
          }
        } catch (err) {
          _iterator5.e(err);
        } finally {
          _iterator5.f();
        }
        ;
      }
      return true;
    }
  }]);
}();
SearchSiteContent200Response.RequiredProperties = ["Articles", "Grocery Products", "Menu Items", "Recipes"];

/**
 * @member {Array.<module:model/SearchResult>} Articles
 */
SearchSiteContent200Response.prototype['Articles'] = undefined;

/**
 * @member {Array.<module:model/SearchResult>} Grocery Products
 */
SearchSiteContent200Response.prototype['Grocery Products'] = undefined;

/**
 * @member {Array.<module:model/SearchResult>} Menu Items
 */
SearchSiteContent200Response.prototype['Menu Items'] = undefined;

/**
 * @member {Array.<module:model/SearchResult>} Recipes
 */
SearchSiteContent200Response.prototype['Recipes'] = undefined;
var _default = exports["default"] = SearchSiteContent200Response;
