// spoonacular/dist/model/SearchRecipesByIngredients200ResponseInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner = _interopRequireDefault(require("./SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _createForOfIteratorHelper(r, e) { var t = "undefined" != typeof Symbol && r[Symbol.iterator] || r["@@iterator"]; if (!t) { if (Array.isArray(r) || (t = _unsupportedIterableToArray(r)) || e && r && "number" == typeof r.length) { t && (r = t); var _n = 0, F = function F() {}; return { s: F, n: function n() { return _n >= r.length ? { done: !0 } : { done: !1, value: r[_n++] }; }, e: function e(r) { throw r; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var o, a = !0, u = !1; return { s: function s() { t = t.call(r); }, n: function n() { var r = t.next(); return a = r.done, r; }, e: function e(r) { u = !0, o = r; }, f: function f() { try { a || null == t["return"] || t["return"](); } finally { if (u) throw o; } } }; }
function _unsupportedIterableToArray(r, a) { if (r) { if ("string" == typeof r) return _arrayLikeToArray(r, a); var t = {}.toString.call(r).slice(8, -1); return "Object" === t && r.constructor && (t = r.constructor.name), "Map" === t || "Set" === t ? Array.from(r) : "Arguments" === t || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(t) ? _arrayLikeToArray(r, a) : void 0; } }
function _arrayLikeToArray(r, a) { (null == a || a > r.length) && (a = r.length); for (var e = 0, n = Array(a); e < a; e++) n[e] = r[e]; return n; }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The SearchRecipesByIngredients200ResponseInner model module.
 * @module model/SearchRecipesByIngredients200ResponseInner
 * @version 2.0.2
 */
var SearchRecipesByIngredients200ResponseInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SearchRecipesByIngredients200ResponseInner</code>.
   * @alias module:model/SearchRecipesByIngredients200ResponseInner
   * @param id {Number} 
   * @param image {String} 
   * @param imageType {String} 
   * @param likes {Number} 
   * @param missedIngredientCount {Number} 
   * @param missedIngredients {Array.<module:model/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner>} 
   * @param title {String} 
   * @param unusedIngredients {Array.<Object>} 
   * @param usedIngredientCount {Number} 
   * @param usedIngredients {Array.<module:model/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner>} 
   */
  function SearchRecipesByIngredients200ResponseInner(id, image, imageType, likes, missedIngredientCount, missedIngredients, title, unusedIngredients, usedIngredientCount, usedIngredients) {
    _classCallCheck(this, SearchRecipesByIngredients200ResponseInner);
    SearchRecipesByIngredients200ResponseInner.initialize(this, id, image, imageType, likes, missedIngredientCount, missedIngredients, title, unusedIngredients, usedIngredientCount, usedIngredients);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  return _createClass(SearchRecipesByIngredients200ResponseInner, null, [{
    key: "initialize",
    value: function initialize(obj, id, image, imageType, likes, missedIngredientCount, missedIngredients, title, unusedIngredients, usedIngredientCount, usedIngredients) {
      obj['id'] = id;
      obj['image'] = image;
      obj['imageType'] = imageType;
      obj['likes'] = likes;
      obj['missedIngredientCount'] = missedIngredientCount;
      obj['missedIngredients'] = missedIngredients;
      obj['title'] = title;
      obj['unusedIngredients'] = unusedIngredients;
      obj['usedIngredientCount'] = usedIngredientCount;
      obj['usedIngredients'] = usedIngredients;
    }

    /**
     * Constructs a <code>SearchRecipesByIngredients200ResponseInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SearchRecipesByIngredients200ResponseInner} obj Optional instance to populate.
     * @return {module:model/SearchRecipesByIngredients200ResponseInner} The populated <code>SearchRecipesByIngredients200ResponseInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SearchRecipesByIngredients200ResponseInner();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
        }
        if (data.hasOwnProperty('image')) {
          obj['image'] = _ApiClient["default"].convertToType(data['image'], 'String');
        }
        if (data.hasOwnProperty('imageType')) {
          obj['imageType'] = _ApiClient["default"].convertToType(data['imageType'], 'String');
        }
        if (data.hasOwnProperty('likes')) {
          obj['likes'] = _ApiClient["default"].convertToType(data['likes'], 'Number');
        }
        if (data.hasOwnProperty('missedIngredientCount')) {
          obj['missedIngredientCount'] = _ApiClient["default"].convertToType(data['missedIngredientCount'], 'Number');
        }
        if (data.hasOwnProperty('missedIngredients')) {
          obj['missedIngredients'] = _ApiClient["default"].convertToType(data['missedIngredients'], [_SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner["default"]]);
        }
        if (data.hasOwnProperty('title')) {
          obj['title'] = _ApiClient["default"].convertToType(data['title'], 'String');
        }
        if (data.hasOwnProperty('unusedIngredients')) {
          obj['unusedIngredients'] = _ApiClient["default"].convertToType(data['unusedIngredients'], [Object]);
        }
        if (data.hasOwnProperty('usedIngredientCount')) {
          obj['usedIngredientCount'] = _ApiClient["default"].convertToType(data['usedIngredientCount'], 'Number');
        }
        if (data.hasOwnProperty('usedIngredients')) {
          obj['usedIngredients'] = _ApiClient["default"].convertToType(data['usedIngredients'], [_SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner["default"]]);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SearchRecipesByIngredients200ResponseInner</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SearchRecipesByIngredients200ResponseInner</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // check to make sure all required properties are present in the JSON string
      var _iterator = _createForOfIteratorHelper(SearchRecipesByIngredients200ResponseInner.RequiredProperties),
        _step;
      try {
        for (_iterator.s(); !(_step = _iterator.n()).done;) {
          var property = _step.value;
          if (!data.hasOwnProperty(property)) {
            throw new Error("The required field `" + property + "` is not found in the JSON data: " + JSON.stringify(data));
          }
        }
        // ensure the json data is a string
      } catch (err) {
        _iterator.e(err);
      } finally {
        _iterator.f();
      }
      if (data['image'] && !(typeof data['image'] === 'string' || data['image'] instanceof String)) {
        throw new Error("Expected the field `image` to be a primitive type in the JSON string but got " + data['image']);
      }
      // ensure the json data is a string
      if (data['imageType'] && !(typeof data['imageType'] === 'string' || data['imageType'] instanceof String)) {
        throw new Error("Expected the field `imageType` to be a primitive type in the JSON string but got " + data['imageType']);
      }
      if (data['missedIngredients']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['missedIngredients'])) {
          throw new Error("Expected the field `missedIngredients` to be an array in the JSON data but got " + data['missedIngredients']);
        }
        // validate the optional field `missedIngredients` (array)
        var _iterator2 = _createForOfIteratorHelper(data['missedIngredients']),
          _step2;
        try {
          for (_iterator2.s(); !(_step2 = _iterator2.n()).done;) {
            var item = _step2.value;
            _SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner["default"].validateJSON(item);
          }
        } catch (err) {
          _iterator2.e(err);
        } finally {
          _iterator2.f();
        }
        ;
      }
      // ensure the json data is a string
      if (data['title'] && !(typeof data['title'] === 'string' || data['title'] instanceof String)) {
        throw new Error("Expected the field `title` to be a primitive type in the JSON string but got " + data['title']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['unusedIngredients'])) {
        throw new Error("Expected the field `unusedIngredients` to be an array in the JSON data but got " + data['unusedIngredients']);
      }
      if (data['usedIngredients']) {
        // data not null
        // ensure the json data is an array
        if (!Array.isArray(data['usedIngredients'])) {
          throw new Error("Expected the field `usedIngredients` to be an array in the JSON data but got " + data['usedIngredients']);
        }
        // validate the optional field `usedIngredients` (array)
        var _iterator3 = _createForOfIteratorHelper(data['usedIngredients']),
          _step3;
        try {
          for (_iterator3.s(); !(_step3 = _iterator3.n()).done;) {
            var _item = _step3.value;
            _SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner["default"].validateJSON(_item);
          }
        } catch (err) {
          _iterator3.e(err);
        } finally {
          _iterator3.f();
        }
        ;
      }
      return true;
    }
  }]);
}();
SearchRecipesByIngredients200ResponseInner.RequiredProperties = ["id", "image", "imageType", "likes", "missedIngredientCount", "missedIngredients", "title", "unusedIngredients", "usedIngredientCount", "usedIngredients"];

/**
 * @member {Number} id
 */
SearchRecipesByIngredients200ResponseInner.prototype['id'] = undefined;

/**
 * @member {String} image
 */
SearchRecipesByIngredients200ResponseInner.prototype['image'] = undefined;

/**
 * @member {String} imageType
 */
SearchRecipesByIngredients200ResponseInner.prototype['imageType'] = undefined;

/**
 * @member {Number} likes
 */
SearchRecipesByIngredients200ResponseInner.prototype['likes'] = undefined;

/**
 * @member {Number} missedIngredientCount
 */
SearchRecipesByIngredients200ResponseInner.prototype['missedIngredientCount'] = undefined;

/**
 * @member {Array.<module:model/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner>} missedIngredients
 */
SearchRecipesByIngredients200ResponseInner.prototype['missedIngredients'] = undefined;

/**
 * @member {String} title
 */
SearchRecipesByIngredients200ResponseInner.prototype['title'] = undefined;

/**
 * @member {Array.<Object>} unusedIngredients
 */
SearchRecipesByIngredients200ResponseInner.prototype['unusedIngredients'] = undefined;

/**
 * @member {Number} usedIngredientCount
 */
SearchRecipesByIngredients200ResponseInner.prototype['usedIngredientCount'] = undefined;

/**
 * @member {Array.<module:model/SearchRecipesByIngredients200ResponseInnerMissedIngredientsInner>} usedIngredients
 */
SearchRecipesByIngredients200ResponseInner.prototype['usedIngredients'] = undefined;
var _default = exports["default"] = SearchRecipesByIngredients200ResponseInner;
