// spoonacular/dist/api/MiscApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _DetectFoodInText200Response = _interopRequireDefault(require("../model/DetectFoodInText200Response"));
var _GetARandomFoodJoke200Response = _interopRequireDefault(require("../model/GetARandomFoodJoke200Response"));
var _GetConversationSuggests200Response = _interopRequireDefault(require("../model/GetConversationSuggests200Response"));
var _ImageAnalysisByURL200Response = _interopRequireDefault(require("../model/ImageAnalysisByURL200Response"));
var _ImageClassificationByURL200Response = _interopRequireDefault(require("../model/ImageClassificationByURL200Response"));
var _SearchAllFood200Response = _interopRequireDefault(require("../model/SearchAllFood200Response"));
var _SearchCustomFoods200Response = _interopRequireDefault(require("../model/SearchCustomFoods200Response"));
var _SearchFoodVideos200Response = _interopRequireDefault(require("../model/SearchFoodVideos200Response"));
var _SearchSiteContent200Response = _interopRequireDefault(require("../model/SearchSiteContent200Response"));
var _TalkToChatbot200Response = _interopRequireDefault(require("../model/TalkToChatbot200Response"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* Misc service.
* @module api/MiscApi
* @version 2.0.2
*/
var MiscApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new MiscApi. 
  * @alias module:api/MiscApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function MiscApi(apiClient) {
    _classCallCheck(this, MiscApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the detectFoodInText operation.
   * @callback module:api/MiscApi~detectFoodInTextCallback
   * @param {String} error Error message, if any.
   * @param {module:model/DetectFoodInText200Response} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Detect Food in Text
   * Take any text and find all mentions of food contained within it. This task is also called Named Entity Recognition (NER). In this case, the entities are foods. Either dishes, such as pizza or cheeseburger, or ingredients, such as cucumber or almonds.
   * @param {String} text 
   * @param {module:api/MiscApi~detectFoodInTextCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/DetectFoodInText200Response}
   */
  return _createClass(MiscApi, [{
    key: "detectFoodInText",
    value: function detectFoodInText(text, callback) {
      var postBody = null;
      // verify the required parameter 'text' is set
      if (text === undefined || text === null) {
        throw new Error("Missing the required parameter 'text' when calling detectFoodInText");
      }
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {
        'text': text
      };
      var authNames = ['apiKeyScheme'];
      var contentTypes = ['application/x-www-form-urlencoded'];
      var accepts = ['application/json'];
      var returnType = _DetectFoodInText200Response["default"];
      return this.apiClient.callApi('/food/detect', 'POST', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getARandomFoodJoke operation.
     * @callback module:api/MiscApi~getARandomFoodJokeCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetARandomFoodJoke200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Random Food Joke
     * Get a random joke that is related to food. Caution: this is an endpoint for adults!
     * @param {module:api/MiscApi~getARandomFoodJokeCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetARandomFoodJoke200Response}
     */
  }, {
    key: "getARandomFoodJoke",
    value: function getARandomFoodJoke(callback) {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetARandomFoodJoke200Response["default"];
      return this.apiClient.callApi('/food/jokes/random', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getConversationSuggests operation.
     * @callback module:api/MiscApi~getConversationSuggestsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetConversationSuggests200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Conversation Suggests
     * This endpoint returns suggestions for things the user can say or ask the chatbot.
     * @param {String} query A (partial) query from the user. The endpoint will return if it matches topics it can talk about.
     * @param {Object} opts Optional parameters
     * @param {Number} [number] The number of suggestions to return (between 1 and 25).
     * @param {module:api/MiscApi~getConversationSuggestsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetConversationSuggests200Response}
     */
  }, {
    key: "getConversationSuggests",
    value: function getConversationSuggests(query, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'query' is set
      if (query === undefined || query === null) {
        throw new Error("Missing the required parameter 'query' when calling getConversationSuggests");
      }
      var pathParams = {};
      var queryParams = {
        'query': query,
        'number': opts['number']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetConversationSuggests200Response["default"];
      return this.apiClient.callApi('/food/converse/suggest', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getRandomFoodTrivia operation.
     * @callback module:api/MiscApi~getRandomFoodTriviaCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetARandomFoodJoke200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Random Food Trivia
     * Returns random food trivia.
     * @param {module:api/MiscApi~getRandomFoodTriviaCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetARandomFoodJoke200Response}
     */
  }, {
    key: "getRandomFoodTrivia",
    value: function getRandomFoodTrivia(callback) {
      var postBody = null;
      var pathParams = {};
      var queryParams = {};
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetARandomFoodJoke200Response["default"];
      return this.apiClient.callApi('/food/trivia/random', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the imageAnalysisByURL operation.
     * @callback module:api/MiscApi~imageAnalysisByURLCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ImageAnalysisByURL200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Image Analysis by URL
     * Analyze a food image. The API tries to classify the image, guess the nutrition, and find a matching recipes.
     * @param {String} imageUrl The URL of the image to be analyzed.
     * @param {module:api/MiscApi~imageAnalysisByURLCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ImageAnalysisByURL200Response}
     */
  }, {
    key: "imageAnalysisByURL",
    value: function imageAnalysisByURL(imageUrl, callback) {
      var postBody = null;
      // verify the required parameter 'imageUrl' is set
      if (imageUrl === undefined || imageUrl === null) {
        throw new Error("Missing the required parameter 'imageUrl' when calling imageAnalysisByURL");
      }
      var pathParams = {};
      var queryParams = {
        'imageUrl': imageUrl
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ImageAnalysisByURL200Response["default"];
      return this.apiClient.callApi('/food/images/analyze', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the imageClassificationByURL operation.
     * @callback module:api/MiscApi~imageClassificationByURLCallback
     * @param {String} error Error message, if any.
     * @param {module:model/ImageClassificationByURL200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Image Classification by URL
     * Classify a food image.
     * @param {String} imageUrl The URL of the image to be classified.
     * @param {module:api/MiscApi~imageClassificationByURLCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/ImageClassificationByURL200Response}
     */
  }, {
    key: "imageClassificationByURL",
    value: function imageClassificationByURL(imageUrl, callback) {
      var postBody = null;
      // verify the required parameter 'imageUrl' is set
      if (imageUrl === undefined || imageUrl === null) {
        throw new Error("Missing the required parameter 'imageUrl' when calling imageClassificationByURL");
      }
      var pathParams = {};
      var queryParams = {
        'imageUrl': imageUrl
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _ImageClassificationByURL200Response["default"];
      return this.apiClient.callApi('/food/images/classify', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the searchAllFood operation.
     * @callback module:api/MiscApi~searchAllFoodCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SearchAllFood200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Search All Food
     * Search all food content with one call. That includes recipes, grocery products, menu items, simple foods (ingredients), and food videos.
     * @param {String} query The search query.
     * @param {Object} opts Optional parameters
     * @param {Number} [offset] The number of results to skip (between 0 and 900).
     * @param {Number} [number = 10)] The maximum number of items to return (between 1 and 100). Defaults to 10.
     * @param {module:api/MiscApi~searchAllFoodCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SearchAllFood200Response}
     */
  }, {
    key: "searchAllFood",
    value: function searchAllFood(query, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'query' is set
      if (query === undefined || query === null) {
        throw new Error("Missing the required parameter 'query' when calling searchAllFood");
      }
      var pathParams = {};
      var queryParams = {
        'query': query,
        'offset': opts['offset'],
        'number': opts['number']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SearchAllFood200Response["default"];
      return this.apiClient.callApi('/food/search', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the searchCustomFoods operation.
     * @callback module:api/MiscApi~searchCustomFoodsCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SearchCustomFoods200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Search Custom Foods
     * Search custom foods in a user's account.
     * @param {String} query The (natural language) search query.
     * @param {String} username The username.
     * @param {String} hash The private hash for the username.
     * @param {Object} opts Optional parameters
     * @param {Number} [offset] The number of results to skip (between 0 and 900).
     * @param {Number} [number = 10)] The maximum number of items to return (between 1 and 100). Defaults to 10.
     * @param {module:api/MiscApi~searchCustomFoodsCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SearchCustomFoods200Response}
     */
  }, {
    key: "searchCustomFoods",
    value: function searchCustomFoods(query, username, hash, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'query' is set
      if (query === undefined || query === null) {
        throw new Error("Missing the required parameter 'query' when calling searchCustomFoods");
      }
      // verify the required parameter 'username' is set
      if (username === undefined || username === null) {
        throw new Error("Missing the required parameter 'username' when calling searchCustomFoods");
      }
      // verify the required parameter 'hash' is set
      if (hash === undefined || hash === null) {
        throw new Error("Missing the required parameter 'hash' when calling searchCustomFoods");
      }
      var pathParams = {};
      var queryParams = {
        'query': query,
        'username': username,
        'hash': hash,
        'offset': opts['offset'],
        'number': opts['number']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SearchCustomFoods200Response["default"];
      return this.apiClient.callApi('/food/customFoods/search', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the searchFoodVideos operation.
     * @callback module:api/MiscApi~searchFoodVideosCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SearchFoodVideos200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Search Food Videos
     * Find recipe and other food related videos.
     * @param {String} query The (natural language) search query.
     * @param {Object} opts Optional parameters
     * @param {String} [type] The type of the recipes. See a full list of supported meal types.
     * @param {String} [cuisine] The cuisine(s) of the recipes. One or more, comma separated. See a full list of supported cuisines.
     * @param {String} [diet] The diet for which the recipes must be suitable. See a full list of supported diets.
     * @param {String} [includeIngredients] A comma-separated list of ingredients that the recipes should contain.
     * @param {String} [excludeIngredients] A comma-separated list of ingredients or ingredient types that the recipes must not contain.
     * @param {Number} [minLength] Minimum video length in seconds.
     * @param {Number} [maxLength] Maximum video length in seconds.
     * @param {Number} [offset] The number of results to skip (between 0 and 900).
     * @param {Number} [number = 10)] The maximum number of items to return (between 1 and 100). Defaults to 10.
     * @param {module:api/MiscApi~searchFoodVideosCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SearchFoodVideos200Response}
     */
  }, {
    key: "searchFoodVideos",
    value: function searchFoodVideos(query, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'query' is set
      if (query === undefined || query === null) {
        throw new Error("Missing the required parameter 'query' when calling searchFoodVideos");
      }
      var pathParams = {};
      var queryParams = {
        'query': query,
        'type': opts['type'],
        'cuisine': opts['cuisine'],
        'diet': opts['diet'],
        'includeIngredients': opts['includeIngredients'],
        'excludeIngredients': opts['excludeIngredients'],
        'minLength': opts['minLength'],
        'maxLength': opts['maxLength'],
        'offset': opts['offset'],
        'number': opts['number']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SearchFoodVideos200Response["default"];
      return this.apiClient.callApi('/food/videos/search', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the searchSiteContent operation.
     * @callback module:api/MiscApi~searchSiteContentCallback
     * @param {String} error Error message, if any.
     * @param {module:model/SearchSiteContent200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Search Site Content
     * Search spoonacular's site content. You'll be able to find everything that you could also find using the search suggestions on spoonacular.com. This is a suggest API so you can send partial strings as queries.
     * @param {String} query The query to search for. You can also use partial queries such as \"spagh\" to already find spaghetti recipes, articles, grocery products, and other content.
     * @param {module:api/MiscApi~searchSiteContentCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/SearchSiteContent200Response}
     */
  }, {
    key: "searchSiteContent",
    value: function searchSiteContent(query, callback) {
      var postBody = null;
      // verify the required parameter 'query' is set
      if (query === undefined || query === null) {
        throw new Error("Missing the required parameter 'query' when calling searchSiteContent");
      }
      var pathParams = {};
      var queryParams = {
        'query': query
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _SearchSiteContent200Response["default"];
      return this.apiClient.callApi('/food/site/search', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the talkToChatbot operation.
     * @callback module:api/MiscApi~talkToChatbotCallback
     * @param {String} error Error message, if any.
     * @param {module:model/TalkToChatbot200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Talk to Chatbot
     * This endpoint can be used to have a conversation about food with the spoonacular chatbot. Use the \"Get Conversation Suggests\" endpoint to show your user what he or she can say.
     * @param {String} text The request / question / answer from the user to the chatbot.
     * @param {Object} opts Optional parameters
     * @param {String} [contextId] An arbitrary globally unique id for your conversation. The conversation can contain states so you should pass your context id if you want the bot to be able to remember the conversation.
     * @param {module:api/MiscApi~talkToChatbotCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/TalkToChatbot200Response}
     */
  }, {
    key: "talkToChatbot",
    value: function talkToChatbot(text, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'text' is set
      if (text === undefined || text === null) {
        throw new Error("Missing the required parameter 'text' when calling talkToChatbot");
      }
      var pathParams = {};
      var queryParams = {
        'text': text,
        'contextId': opts['contextId']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _TalkToChatbot200Response["default"];
      return this.apiClient.callApi('/food/converse', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
}();
