// spoonacular/dist/api/WineApi.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _GetDishPairingForWine200Response = _interopRequireDefault(require("../model/GetDishPairingForWine200Response"));
var _GetWineDescription200Response = _interopRequireDefault(require("../model/GetWineDescription200Response"));
var _GetWinePairing200Response = _interopRequireDefault(require("../model/GetWinePairing200Response"));
var _GetWineRecommendation200Response = _interopRequireDefault(require("../model/GetWineRecommendation200Response"));
function _interopRequireDefault(e) { return e && e.__esModule ? e : { "default": e }; }
function _typeof(o) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (o) { return typeof o; } : function (o) { return o && "function" == typeof Symbol && o.constructor === Symbol && o !== Symbol.prototype ? "symbol" : typeof o; }, _typeof(o); }
function _classCallCheck(a, n) { if (!(a instanceof n)) throw new TypeError("Cannot call a class as a function"); }
function _defineProperties(e, r) { for (var t = 0; t < r.length; t++) { var o = r[t]; o.enumerable = o.enumerable || !1, o.configurable = !0, "value" in o && (o.writable = !0), Object.defineProperty(e, _toPropertyKey(o.key), o); } }
function _createClass(e, r, t) { return r && _defineProperties(e.prototype, r), t && _defineProperties(e, t), Object.defineProperty(e, "prototype", { writable: !1 }), e; }
function _toPropertyKey(t) { var i = _toPrimitive(t, "string"); return "symbol" == _typeof(i) ? i : i + ""; }
function _toPrimitive(t, r) { if ("object" != _typeof(t) || !t) return t; var e = t[Symbol.toPrimitive]; if (void 0 !== e) { var i = e.call(t, r || "default"); if ("object" != _typeof(i)) return i; throw new TypeError("@@toPrimitive must return a primitive value."); } return ("string" === r ? String : Number)(t); } /**
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 2.0.2
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
* Wine service.
* @module api/WineApi
* @version 2.0.2
*/
var WineApi = exports["default"] = /*#__PURE__*/function () {
  /**
  * Constructs a new WineApi. 
  * @alias module:api/WineApi
  * @class
  * @param {module:ApiClient} [apiClient] Optional API client implementation to use,
  * default to {@link module:ApiClient#instance} if unspecified.
  */
  function WineApi(apiClient) {
    _classCallCheck(this, WineApi);
    this.apiClient = apiClient || _ApiClient["default"].instance;
  }

  /**
   * Callback function to receive the result of the getDishPairingForWine operation.
   * @callback module:api/WineApi~getDishPairingForWineCallback
   * @param {String} error Error message, if any.
   * @param {module:model/GetDishPairingForWine200Response} data The data returned by the service call.
   * @param {String} response The complete HTTP response.
   */

  /**
   * Dish Pairing for Wine
   * Find a dish that goes well with a given wine.
   * @param {String} wine The type of wine that should be paired, e.g. \"merlot\", \"riesling\", or \"malbec\".
   * @param {module:api/WineApi~getDishPairingForWineCallback} callback The callback function, accepting three arguments: error, data, response
   * data is of type: {@link module:model/GetDishPairingForWine200Response}
   */
  return _createClass(WineApi, [{
    key: "getDishPairingForWine",
    value: function getDishPairingForWine(wine, callback) {
      var postBody = null;
      // verify the required parameter 'wine' is set
      if (wine === undefined || wine === null) {
        throw new Error("Missing the required parameter 'wine' when calling getDishPairingForWine");
      }
      var pathParams = {};
      var queryParams = {
        'wine': wine
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetDishPairingForWine200Response["default"];
      return this.apiClient.callApi('/food/wine/dishes', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getWineDescription operation.
     * @callback module:api/WineApi~getWineDescriptionCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetWineDescription200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Wine Description
     * Get a simple description of a certain wine, e.g. \"malbec\", \"riesling\", or \"merlot\".
     * @param {String} wine The name of the wine that should be paired, e.g. \"merlot\", \"riesling\", or \"malbec\".
     * @param {module:api/WineApi~getWineDescriptionCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetWineDescription200Response}
     */
  }, {
    key: "getWineDescription",
    value: function getWineDescription(wine, callback) {
      var postBody = null;
      // verify the required parameter 'wine' is set
      if (wine === undefined || wine === null) {
        throw new Error("Missing the required parameter 'wine' when calling getWineDescription");
      }
      var pathParams = {};
      var queryParams = {
        'wine': wine
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetWineDescription200Response["default"];
      return this.apiClient.callApi('/food/wine/description', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getWinePairing operation.
     * @callback module:api/WineApi~getWinePairingCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetWinePairing200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Wine Pairing
     * Find a wine that goes well with a food. Food can be a dish name (\"steak\"), an ingredient name (\"salmon\"), or a cuisine (\"italian\").
     * @param {String} food The food to get a pairing for. This can be a dish (\"steak\"), an ingredient (\"salmon\"), or a cuisine (\"italian\").
     * @param {Object} opts Optional parameters
     * @param {Number} [maxPrice] The maximum price for the specific wine recommendation in USD.
     * @param {module:api/WineApi~getWinePairingCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetWinePairing200Response}
     */
  }, {
    key: "getWinePairing",
    value: function getWinePairing(food, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'food' is set
      if (food === undefined || food === null) {
        throw new Error("Missing the required parameter 'food' when calling getWinePairing");
      }
      var pathParams = {};
      var queryParams = {
        'food': food,
        'maxPrice': opts['maxPrice']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetWinePairing200Response["default"];
      return this.apiClient.callApi('/food/wine/pairing', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }

    /**
     * Callback function to receive the result of the getWineRecommendation operation.
     * @callback module:api/WineApi~getWineRecommendationCallback
     * @param {String} error Error message, if any.
     * @param {module:model/GetWineRecommendation200Response} data The data returned by the service call.
     * @param {String} response The complete HTTP response.
     */

    /**
     * Wine Recommendation
     * Get a specific wine recommendation (concrete product) for a given wine type, e.g. \"merlot\".
     * @param {String} wine The type of wine to get a specific product recommendation for.
     * @param {Object} opts Optional parameters
     * @param {Number} [maxPrice] The maximum price for the specific wine recommendation in USD.
     * @param {Number} [minRating] The minimum rating of the recommended wine between 0 and 1. For example, 0.8 equals 4 out of 5 stars.
     * @param {Number} [number = 10)] The number of wine recommendations expected (between 1 and 100).
     * @param {module:api/WineApi~getWineRecommendationCallback} callback The callback function, accepting three arguments: error, data, response
     * data is of type: {@link module:model/GetWineRecommendation200Response}
     */
  }, {
    key: "getWineRecommendation",
    value: function getWineRecommendation(wine, opts, callback) {
      opts = opts || {};
      var postBody = null;
      // verify the required parameter 'wine' is set
      if (wine === undefined || wine === null) {
        throw new Error("Missing the required parameter 'wine' when calling getWineRecommendation");
      }
      var pathParams = {};
      var queryParams = {
        'wine': wine,
        'maxPrice': opts['maxPrice'],
        'minRating': opts['minRating'],
        'number': opts['number']
      };
      var headerParams = {};
      var formParams = {};
      var authNames = ['apiKeyScheme'];
      var contentTypes = [];
      var accepts = ['application/json'];
      var returnType = _GetWineRecommendation200Response["default"];
      return this.apiClient.callApi('/food/wine/recommendation', 'GET', pathParams, queryParams, headerParams, formParams, postBody, authNames, contentTypes, accepts, returnType, null, callback);
    }
  }]);
}();
